src/reflex.cpp

```diff
@@ -1839,7 +1868,36 @@ void Reflex::write_class()
       "    :\n"
       "      " << base << "(input, os)\n";
     write_section_init();
-    if (!options["bison_cc"].empty())
+    if (!options["bison_complete"].empty())
+    {
+      const std::string& ret_type = options["token_type"];
+      if (!options["bison_locations"].empty())
```

Comment:
I'm not sure I understand this. When `bison-locations` is set, we generate a `yylloc_update(yylloc)` function that is called by `yylex()`, see `write_lexer` at the original line 2447. Is this change valid and can we still make that call there?

Reply:
According to [the Bison manual](https://www.gnu.org/software/bison/manual/html_node/Complete-Symbols.html#Complete-Symbols), if `%locations` is used, the generated `symbol_type` accept a `location` parameter in its constructor. (See below)

> For each token type, Bison generates named constructors as follows.
> Constructor on parser::symbol_type: symbol_type (int token, const value_type& value, const location_type& location)
> Constructor on parser::symbol_type: symbol_type (int token, const location_type& location)
> Constructor on parser::symbol_type: symbol_type (int token, const value_type& value)
> Constructor on parser::symbol_type: symbol_type (int token)
> Build a complete terminal symbol for the token type token (including the api.token.prefix), whose semantic value, if it has one, is value of adequate value_type. _**Pass the location iff location tracking is enabled.**_

Since RE-flex does not allow custom parameters for `lex`, this implies that there are no parameters at all. 

Now that there is no `yylloc` parameter, we cannot call `yylloc_update` on it, so I removed that function call, modified that function and renamed it as `location`. As I showed in the test case in `bison-complete-test/scanner.l`, users can call `location()` to get the `location` object to pass on to the token constructor.

```c++
return make_TOKEN(some_semantic_values, location());
```